packages/aws-cdk-lib/aws-rds/README.md

```diff
@@ -1166,30 +1166,31 @@ new rds.ServerlessClusterFromSnapshot(this, 'Cluster', {
 
 ### Data API
 
-You can access your Aurora Serverless DB cluster using the built-in Data API. The Data API doesn't require a persistent connection to the DB cluster. Instead, it provides a secure HTTP endpoint and integration with AWS SDKs.
+You can access your Aurora DB cluster using the built-in Data API. The Data API doesn't require a persistent connection to the DB cluster. Instead, it provides a secure HTTP endpoint and integration with AWS SDKs.
 
 The following example shows granting Data API access to a Lamba function.
 
 ```ts
 declare const vpc: ec2.Vpc;
+declare const fn: lambda.Function;
 
-const cluster = new rds.ServerlessCluster(this, 'AnotherCluster', {
+// Create a serverless V1 cluster
+const serverlessV1Cluster = new rds.ServerlessCluster(this, 'AnotherCluster', {
   engine: rds.DatabaseClusterEngine.AURORA_MYSQL,
   vpc, // this parameter is optional for serverless Clusters
   enableDataApi: true, // Optional - will be automatically set if you call grantDataApiAccess()
 });
+serverlessV1Cluster.grantDataApiAccess(fn);
 
-declare const code: lambda.Code;
-const fn = new lambda.Function(this, 'MyFunction', {
-  runtime: lambda.Runtime.NODEJS_LATEST,
-  handler: 'index.handler',
-  code,
-  environment: {
-    CLUSTER_ARN: cluster.clusterArn,
-    SECRET_ARN: cluster.secret!.secretArn,
-  },
+// Create an Aurora cluster
+const cluster = new rds.DatabaseCluster(this, 'Cluster', {
+  engine: rds.DatabaseClusterEngine.AURORA_MYSQL,
+  vpc,
+  enableDataApi: true, // Optional - will be automatically set if you call grantDataApiAccess()
 });
 cluster.grantDataApiAccess(fn);
+// It is necessary to grant the function access to the secret associated with the cluster for `DatabaseCluster`.
+cluster.secret!.grantRead(fn);
```

Comment:
Curious why is this necessary for Aurora cluster but not for the serverless cluster?

Reply:
@GavinZZ 

In the serverless cluster, `cluster.secret` is defined in the `ServerlessClusterBase` class (see [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-rds/lib/serverless-cluster.ts#L341)), and `secret.grantRead()` is invoked within `ServerlessClusterBase.grantDataApiAccess()`.

However, for the Aurora cluster, `cluster.secret` is not defined in `DatabaseClusterBase` class but rather in `DatabaseClusterNew` class (refer to [this link](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-rds/lib/cluster.ts#L526)).

To align these implementations, it's necessary to move the definition of `cluster.secret` from `DatabaseClusterNew` to `DatabaseClusterBase`. I anticipate this change might be significant. Therefore, I plan to open an issue regarding this modification after the current PR is merged.

I can attempt to make this modification within the current PR if you prefer. What are your thoughts on this?

Reply:
Ah I see, it makes sense here to split it into two PRs.